### PR TITLE
Deprecate Suite7 & SuiteR Themes from themedef.php

### PR DESCRIPTION
**Pull request by gymad**
## Description

Deprecate Suite7 & SuiteR Themes from themedef.php
(ref: scrm-333)
- acceptable suitecrm version added to themedef.php

## How To Test This

- add/remove/modify value of $themedef['version']['regex_matches'] = array('^7\.[0-8][^\d]'); from themedef.php
- update $suitecrm_version (for e.g upgrade or just change it in suitecrm_version.php)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [ ] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [ ] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: themes/SuiteP/themedef.php [hunk 1]
```diff
@@ -47,6 +47,7 @@
     'description' => 'SuiteCRM Responsive Theme',
     'version' => array(
         'regex_matches' => array('6\.*.*'),
```
Comment (mattlorimer) [inline themes/SuiteP/themedef.php]: it should just use regex_matches no need for regex_matches_acceptable_suitecrm

File: include/SugarTheme/SugarTheme.php [hunk 2]
```diff
@@ -1196,25 +1196,38 @@ public static function add(
         )
     {
         // make sure the we know the sugar version
-        global $sugar_version;
+        global $sugar_version, $suitecrm_version;
```
Comment (mattlorimer) [inline include/SugarTheme/SugarTheme.php]: only need to check suitecrm version

**Comment by pgorod** [comment]
SuiteR is going to be deprecated? What will happen to people using it, they'll just get blocked out of it and forced to use SuiteP? Starting when?

**Comment by pribeiro42** [comment]
The way I see it I think there's a mismatch between the title and the   proposed changes...

**Comment by samus-aran** [comment]
Hi Guys, just here to address your concerns.

- We are not supporting i.e. maintaining SuiteR/Suite7 themes going forward.
- Users can still use SuiteR/Suite7 if they set their instances themesdefs to accept 7.9+
- Upon upgrade Users will default to SuiteP and SuiteR/Suite7 will not be enabled (unless they change the themedefs).

This is scheduled from 7.9+

**Comment by rowlandkeable** [comment]
Sorry but where do I find themedefs so that I can set it to allow SuiteR as all my users prefer the SuiteR theme.  The style of themeP is good but the colours are too washed out and drab looking so I am sorry but my users refuse to use it.

Rowland


Rowland Keable
ICT Administrator
illuma Lighting
11 Sills Road
Willow Farm Business Park
Castle Donington
Derbyshire
DE74 2US
Tel: 01332 818232
Mob: 07720970415
Email: rk@illuma.co.uk
Web: www.illuma.co.uk


This email and any files transmitted with it are confidential and intended solely for the use of the individual or entity to whom they are addressed. If you have received this email in error please notify the sender and delete the e-mail entirely from your system. Please note that any views or opinions presented in this email are solely those of the author and do not necessarily represent those of the company. The recipient should check this email and any attachments for the presence of viruses. The company accepts no liability for any damage caused by any virus transmitted by this email.

**Comment by pgorod** [comment]
I'm really disappointed with these news. I say them coming, but I do think it's too soon. 

At least **the UI to customize SuiteP** would have to be in place, so people can _easily_ correct what they don't like in SuiteP (color, spacing, etc., depending on personal tastes), **one version before deprecation**, to give us time to adjust. It's a sensitive issue with users.

Why is this coming now? Is it because of the changes to the email client being too complicated to do in two themes?

**Comment by samus-aran** [comment]
Hi @rowlandkeable We'll release the steps to allow users to continue using the older themes closer to the time of release.

@pogrod Essentially yes, maintaining three seperate themes is a huge headache especially when we want to push forward with technology.  Regards to the customisations, you are able to change the padding via sass, but are you meaning the colours similar to how SuiteR theme settings are?

**Comment by pgorod** [comment]
@samus-aran my experience changing the SASS is not very positive - granted, it's because I'm no good at CSS and I don't like it : - )

However, the bottom-line is that changing the padding is a complex job that involves a lot of tweaking. Not many people are going to venture into that. I've been asking (#2288) for anybody that did it, to give me the SASS, but I don't think anybody really did it yet.

It should be a mode (compact / spaced) or (fingers / mouse) or (mobile / desktop) or whatever is the best name for it. A switch with two or three options.

And then, yes, the color scheme like on SuiteR would be useful. @daniel-samson said he had started work on a UI to change the SASS, but I see he's been quite busy with other things.

Themes are no longer themes in SuiteCRM. They're too tightly coupled to what's underneath. They are front-ends. That's why it's complicated to maintain three when the technology has changed in one of them. Remember the old SugarCRM, with the buttons on the top bar for the end user to change color palettes? Remember how many themes you could find online? Those were real Themes, just eye-candy, much easier to make.

I praise the technology changes you've been doing, and I understand now is not a good time to try and redo the architecture. I just think you need to insure _a minimum_ of visual configurability without going into CSS and code.

P.S - I have a suggestion: a poll on the forums asking which theme people are using, and if it's not SuiteP, ask why they didn't change. I think (though it's just a hunch, I don't really know) you will be surprised at the results. Anyway it would be useful to guide decisions on Theme deprecation.

**Comment by samus-aran** [comment]
To everyone - we are closing this simply because of conflicts in branches.  The discussion will be moved to forums for further discussions - once this forum thread has been set up I'll link it back to this PR.

**Comment by samus-aran** [comment]
Discussion moved to forums:

https://suitecrm.com/forum/suggestion-box/13393-deprecation-of-suiter-suite7-themes-discussion-on-adopting-suitep

**Pull request closed without merging**